src/App.tsx

```diff
@@ -17,18 +40,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                searchQuery={searchQuery}
+                setSearchQuery={setSearchQuery}
+                progress={progress}
+                setProgress={setProgress}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              <TodoList
+                todos={todosToRender}
+                selectedTodo={selectedTodo}
+                setSelectedTodo={setSelectedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo !== null
```

Comment:
if your selectedTodo is null than you can write !selectedTodo

Reply:
> if your selectedTodo is null than you can write !selectedTodo

sure, i was rewritting logic, coz some thigns didn`t work properly and tried to resolve dat problem in different ways, but forgot bout that.

Reply:
> if your selectedTodo is null than you can write !selectedTodo

Good point 👍  But it should be `selectedTodo` as it doesn't make sense to render modal if there is no selected todo, I think you just accidentally added `!`